### PR TITLE
Added sampling frequency and test for HS2 and other stuff [WIP]

### PR DESCRIPTION
**Pull request by colehurwitz**
Adding some sampling frequency stuff in this pull request

### CODE REVIEW

**Comment by colehurwitz** [comment]
@samuelgarcia @alejoe91  This implements sampling frequency for numpy sorting exactor, subsorting extractor, multisortingextractor, curationextractor, npz, and hs2. It also adds to the tests to start testing for this (added hs2, npz, multisorting, subsorting, curationsorting, tests)

**Comment by alejoe91** [comment]
ready to merge @colehurwitz ?

**Comment by colehurwitz** [comment]
Yes, ready to merge I think. Do you approve of the change to numpy sorting extractor?

File: spikeextractors/extractors/npzsortingextractor/npzsortingextractor.py [hunk 1]
```diff
@@ -63,12 +63,11 @@ def write_sorting(sorting, save_file):
...
-        
-        #~ if sorting._sampling_frequency is not None:
-            #~ d['sampling_frequency'] = np.array([self._sampling_frequency], dtype='float64')
-        
-        np.savez(save_file, **d)
 
+        if sorting.get_sampling_frequency() is not None:
+            d['sampling_frequency'] = np.array([sorting.get_sampling_frequency()], dtype='float64')
+
+        np.savez(save_file, **d)
```
Comment (samuelgarcia) [inline spikeextractors/extractors/npzsortingextractor/npzsortingextractor.py]: Look good!!

**Comment by samuelgarcia** [comment]
This looks good.

**Pull request merged**